app/controllers/checkout_controller.rb

```diff
@@ -142,7 +142,7 @@ def valid_payment_intent_provided?
   end
 
   def handle_redirect_from_stripe
-    if OrderWorkflow.new(@order).next && order_complete?
+    if OrderWorkflow.new(@order).handle_redirect_from_stripe
```

Comment:
This might need looking at (and more specs)... the pre-extracted version is equivalent though (eg: `next && order_complete? && @order.process_payments!`).

What happens if/when payment processing fails after order completion here? Lets imagine:
- We do `OrderWorkflow.new(@order).next`
- The order is completed
- We then do `@order.process_payments!` but it fails (for whatever reason)

What happens next?

- We jump to the other side of this conditional and call `checkout_failed`...
- which calls `Checkout::PostCheckoutActions.new(@order).failure`...
- which calls `OrderCheckoutRestart.new(@order).call`...

and then we'll get to here:
https://github.com/openfoodfoundation/openfoodnetwork/blob/a1317be19b76d1953e38f9e47cbc8e5304d275e6/app/services/order_checkout_restart.rb#L9-L17

and then line 12 will throw a `StateMachines::InvalidTransition` error, which won't be caught by the controller action?

Reply:
That's definitely what happened, I verified that I got a state machine transition error when I failed SCA auth on a payment. The latest commit should avoid this.